pywr/core.py

```diff
@@ -118,9 +132,22 @@ def __init__(self, solver=None, parameters=None):
 
         self.node = {}
         self.group = {}
+        self.scenarios = ScenarioCollection()
 
         self.reset()
 
+    @property
+    def number_of_timesteps(self):
+        return len(self.timestepper)
+
+    @property
+    def number_of_scenarios(self):
```

Comment:
`model.scenaios.combinations` would be an iterable object with `__len__` implemented, so that you could do:

```
length = len(model.scenarios.combinations)
for combination in model.scenarios.combinations:
    pass
```

It would require a small refactor of `cartesian` so that it was iterable, but I don't think that should be too hard.